clients/client-python/gravitino/filesystem/gvfs_base_operations.py

```diff
@@ -0,0 +1,497 @@
+            )
+        return pre_processed_path
+
+    def _assert_same_fileset(self, gvfs_path1: str, gvfs_path2: str):
+        """Assert that the two gvfs paths are the same fileset.
+        :param gvfs_path1: The first gvfs path
+        :param gvfs_path2: The second gvfs path
+        """
+        src_path = self._pre_process_path(gvfs_path1)
+        dst_path = self._pre_process_path(gvfs_path2)
+        src_identifier: NameIdentifier = extract_identifier(self._metalake, src_path)
+        dst_identifier: NameIdentifier = extract_identifier(self._metalake, dst_path)
+        if src_identifier != dst_identifier:
+            raise GravitinoRuntimeException(
+                f"Destination file path identifier: `{dst_identifier}` should be same with src file path "
+                f"identifier: `{src_identifier}`."
+            )
+
+    def _init_current_location_name(self):
+        """Initialize the current location name.
+         get from configuration first, otherwise use the env variable
+         if both are not set, return null which means use the default location
+        :return: The current location name
+        """
+        current_location_name_env_var = (
+            self._options.get(GVFSConfig.GVFS_FILESYSTEM_CURRENT_LOCATION_NAME_ENV_VAR)
+            if self._options
+            else None
+        ) or self.ENV_CURRENT_LOCATION_NAME_ENV_VAR_DEFAULT
+
+        return (
+            self._options.get(GVFSConfig.GVFS_FILESYSTEM_CURRENT_LOCATION_NAME)
+            if self._options
+            else None
+        ) or os.environ.get(current_location_name_env_var)
+
+    def _get_actual_filesystem(
+        self, gvfs_path: str, location_name: str
+    ) -> AbstractFileSystem:
+        """Get the actual filesystem based on the gvfs path.
+        :param gvfs_path: The gvfs path
+        :param location_name: The location name, None means the default location
+        :return: The actual filesystem
+        """
+        fileset_ident: NameIdentifier = extract_identifier(self._metalake, gvfs_path)
+        return self._get_actual_filesystem_by_location_name(
+            fileset_ident, location_name
+        )
+
+    def _get_actual_filesystem_by_location_name(
+        self, fileset_ident: NameIdentifier, location_name: str
+    ) -> AbstractFileSystem:
+        """Get the actual filesystem based on the fileset identifier and location name.
+        :param fileset_ident: The fileset identifier
+        :param location_name: The location name, None means the default location
+        :return: The actual filesystem
+        """
+        catalog_ident: NameIdentifier = NameIdentifier.of(
+            self._metalake, fileset_ident.namespace().level(1)
+        )
+        catalog = self._get_fileset_catalog(catalog_ident)
+        fileset = catalog.as_fileset_catalog().load_fileset(
+            NameIdentifier.of(fileset_ident.namespace().level(2), fileset_ident.name())
```

Comment:
Where do we handle the exceptions thrown here?

Reply:
the decorater `_with_exception_translation` in `gvfs.py` will handle it. such as: https://github.com/apache/gravitino/pull/6964/files#diff-0dacbe325e9f4949d30bbd4370ce66c653263e5d4fb47ee09febb1795f0d82a6R139